Added support for Nucleo F042K6

cajt opened the pull request:
It is based on the BSP for Nucelo F031.
Testing is mostly incomplete. I only tested: USART1 & USART2 as loggers, LED blinking.
Basically, I fiddled the clock generation to make the USARTs work, and fixed things along the way.

salkinium reviewed (dismissed):
Thanks!

salkinium commented:
Can you rebase your commit on top of the current develop?
Unfortunately you didn't use a feature branch, so if I did the cherry-pick for you and force pushed this to your develop branch, it would be out of sync with your local branch.
I wish there was a GitHub option to prevent pull requests on the develop Branch.

cajt committed:
Added support for Nucleo F042K6

It is based on the BSP for Nucelo F031.
Testing is mostly incomplete. I only tested: USART1 & USART2 as loggers, LED blinking.
Basically, I fiddled the clock generation to make the USARTs work, and fixed things along the way.

salkinium commented:
ProTip: Don't review late at night. I modified some things incl. adding the example into the CI.

salkinium commented:
FYI: I force pushed the develop branch to your repo, you'll need to `git reset --hard origin/develop` on your local machine. Please use a `feature/name` branch in future to avoid getting `develop` out of sync.
